hack/ci/link-check.sh

```diff
@@ -2,8 +2,10 @@
 set -ev
 
 CONTAINER_RUN_EXTRA_OPTIONS=${CONTAINER_RUN_EXTRA_OPTIONS:=""}
+CONTAINER_ENGINE=${CONTAINER_ENGINE:="docker"}
 
-docker volume create olm-html
-docker run --rm ${CONTAINER_RUN_EXTRA_OPTIONS} -v "$(pwd):/src" -v olm-html:/src/public klakegg/hugo:0.73.0-ext-ubuntu
-docker run --rm -v olm-html:/target mtlynch/htmlproofer /target --empty-alt-ignore --http-status-ignore 429 --allow_hash_href
-docker volume rm olm-html
+# TODO(tflannag): We may need to trap that `... volume rm` call.
```

Comment:
I can remove this TODO in a second. Just need to verify what happens when this script fails might way through. I would guess it would result in this volume still being present despite the script exiting.

Reply:
Yeah played around with this locally - we definitely need to trap that exit signal and cleanup the volume that was created. Added another commit that upgrades the linting script to trap and call a cleanup function.